src/Battle.php

```diff
@@ -0,0 +1,211 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core;
+
+use LotGD\Core\{
+    DiceBag,
+    Exceptions\ArgumentException,
+    Exceptions\BattleIsOverException,
+    Exceptions\BattleNotOverException,
+    Models\FighterInterface
+};
+
+/**
+ * Description of Battle
+ */
+class Battle
+{
+    const DAMAGEROUND_PLAYER = 0b01;
+    const DAMAGEROUND_MONSTER = 0b10;
+    const DAMAGEROUND_BOTH = 0b11;
+    
+    protected $player;
+    protected $monster;
+    protected $diceBag;
+    protected $isOver = false;
+    protected $winner;
+    protected $looser;
+    
+    public function __construct(FighterInterface $player, FighterInterface $monster)
```

Comment:
Pass the `Game` into this constructor, then extract the `DiceBag` from there (so create a `DiceBag` as part of the `Game` setup).